### DISCUSSION TITLE
Remove unused tags

### DISCUSSION THREAD

**Discussion by billsargent** [gotson/komga#1698] [Q&A] [answered]
This is only a feature request IF its not already implemented.... I seem to have tags that lead nowhere. Since I've imported massive amounts of manga and then deleted based on tags, the tags still seem to be there.

**Comment by gotson** [comment]
Can you explain why you think tags lead to nowhere ? Komga does not store tags separately, they are on a book or series, thus having tags leading to nowhere is not possible

**Comment by billsargent** [comment]
You keep tags that were scanned in within an sqlite database,right? What happens when books or series are deleted manually outside of komga? Are they deleted when the libraries are rescanned and trash emptied? I have had tags that showed no results when I did that.  So I'm asking.

**Reply by gotson** [reply, marked as answer]
> You keep tags that were scanned in within an sqlite database,right?

no.

> What happens when books or series are deleted manually outside of komga?

series/books are put in the trash if the file on disk does not exist anymore. If you empty the trash, data is deleted.

> I have had tags that showed no results when I did that. So I'm asking.

could be a cache issue in the browser, try refreshing.

**Reply by billsargent** [reply]
> could be a cache issue in the browser, try refreshing.

Ever had one of those feelings where you suddenly think "I'm a giant dumbass"? Yeah.... that was exactly it. I figured it out yesterday.  I'm sorry I wasted your time. :)

**Reply by gotson** [reply]
The Dashboard loads the tags on startup, so indeed it could happen that some tags are still in the Dashboard cache and not used anymore after some deletions.

It's way to much data to reload the tags all the time, or even to reload them after a series/book changed.

**Reply by billsargent** [reply]
During start up does the cache clear?

**Reply by gotson** [reply]
> During start up does the cache clear?

every time the webpage is loaded, the cache is on client side, not server side

**Reply by billsargent** [reply]
Ok your wording made think you meant a cache in komga itself, not a cached page in the browser.

**Reply by gotson** [reply]
> Ok your wording made think you meant a cache in komga itself, not a cached page in the browser.

it's not a cached page in the browser, it's a stored list of tags in javascript in the Single Page Application that runs in the browser.

**Reply by billsargent** [reply]
OK yes you're right. I forget that its actually a ton of javascript running and that is not serverside and that this isn't just static html. Still working my way through coffee.
